README.md

```diff
@@ -1,5 +1,120 @@
 ﻿# MagicBytesValidator
 
 ### How to install?
+
+Install nuget package into your project
+
+```powershell
+Install-Package MagicBytesValidator -Version 1.0.0
+```
+
+```powershell
+dotnet add package MagicBytesValidator --version 1.0.0
+```
+
+```powershell
+<PackageReference Include="MagicBytesValidator" Version="1.0.0" />
+```
+
 ### How to use it?
+
+Create a new instance of the Validator
+
+```c#
+var validator = new Validator(); //or initialize it with DI
+```
+
+Check a file with the stream and the filetype
+
+```c#
+var isValid = _validator.IsValidAsync(memoryStream, fileType);
+```
+
+Create a filetype by extension or mimetype
+
+```c#
+var pngileType = validator.Mapping.FindByExtension("png");
+var pdfFileType = validator.Mapping.FindByMimeType("application/pdf");
+```
+
+### Expand standard the filetype list
+
+you can add or register your own filetypes. Use the validator or create an instance of the Mapper.
+
+```c#
+var mapper = new Mapping();
+```
+
+Register a single Filetype
+
+```c#
+mapper.Register(
+_trpFileType = new FileType("traperto/trp", new[] { "trp" },
+                                        new[] { new byte[] { 0x74, 0x72, 0x61, 0x70, 0x65, 0x72, 0x74, 0x6f } });
+)
+```
+
+you can also register a list of filetypes
+
+```c#
+mapper.Register( listOfFileTypes );
+```
+
+you can register new fileTypes by an assembly
```

Comment:
Würde ich nock weiter beschreiben, z.B. "You can also create variants of `FileType` and register them by passing the Assembly of the new `FileType`s, e.g. `mapping.Register(typeof(CustomFileType).Assembly);`. This will register _all_ `FileType`s of the given Assembly that are also not abstract and have an empty constructor!"